dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/text/BeamTextCSVTableIOReader.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.dsls.sql.schema.text;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.StringReader;
+
+import org.apache.beam.dsls.sql.schema.BeamSQLRecordType;
+import org.apache.beam.dsls.sql.schema.BeamSQLRow;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * IOReader for {@code BeamTextCSVTable}.
+ */
+public class BeamTextCSVTableIOReader
+    extends PTransform<PBegin, PCollection<BeamSQLRow>>
+    implements Serializable {
+  private static final Logger LOG = LoggerFactory.getLogger(BeamTextCSVTableIOReader.class);
+  private String filePattern;
+  protected BeamSQLRecordType beamSqlRecordType;
+  protected CSVFormat csvFormat;
+
+  public BeamTextCSVTableIOReader(BeamSQLRecordType beamSqlRecordType, String filePattern,
+      CSVFormat csvFormat) {
+    this.filePattern = filePattern;
+    this.beamSqlRecordType = beamSqlRecordType;
+    this.csvFormat = csvFormat;
+  }
+
+  @Override
+  public PCollection<BeamSQLRow> expand(PBegin input) {
+    return input.apply("decodeRecord", TextIO.Read.from(filePattern))
+        .apply(ParDo.of(new DoFn<String, BeamSQLRow>() {
+          @ProcessElement
+          public void processElement(ProcessContext ctx) {
+            String str = ctx.element();
+
+            StringReader reader = new StringReader(str);
+            CSVRecord rawRecord = null;
+            try {
+              CSVParser parser = csvFormat.parse(reader);
+              rawRecord = parser.getRecords().get(0);
+            } catch (IOException e) {
+              LOG.error("error record: " + str, e);
```

Comment:
I was thinking that some error records should not fail the whole job. But lets be strict for now.